src/utils/token.ts

```diff
@@ -0,0 +1,26 @@
+import { parseWarpRouteMessage } from '@hyperlane-xyz/utils';
+import { utils } from 'ethers';
+import { postgresByteaToAddress } from '../features/messages/queries/encoding';
+import { WarpRouteDetails } from '../types';
+
+export function parseWarpRouteDetails(
```

Comment:
Please move this to `src/features/messages/queries/parse.ts` as that's the only place it's used.
I'm not a big fan of accumulating business logic in the `utils` folders